pkg/policyprovider/configpolicy/configpolicy.go

```diff
@@ -74,3 +104,43 @@ func (enforcer PolicyEnforcer) ErrorToVerifyResult(ctx context.Context, subjectR
 	reports = append(reports, errorReport)
 	return types.VerifyResult{IsSuccess: false, VerifierReports: reports}
 }
+
+// OverallVerifyResult determines the final outcome of verification that is constructed using the results from
+// individual verifications
+func (enforcer PolicyEnforcer) OverallVerifyResult(ctx context.Context, verifierReports []interface{}) bool {
+	// use boolean map to track if each artifact type policy constraint is satisfied
+	verifySuccess := map[string]bool{}
+	for artifactType := range enforcer.ArtifactTypePolicies {
+		// add all policies except for default
+		if artifactType != defaultPolicyName {
+			verifySuccess[artifactType] = false
+		}
+	}
+
+	for _, report := range verifierReports {
+		castedReport := report.(verifier.VerifierResult)
+		// the artifact type of the verified artifact matches type specified in policy
+		if policyType, ok := enforcer.ArtifactTypePolicies[castedReport.ArtifactType]; ok {
+			// the artifact type of the verified artifact matches type specified in policy
+			if policyType == vt.AnyVerifySuccess && castedReport.IsSuccess {
+				// if policy is 'any' and report is successful
+				verifySuccess[castedReport.ArtifactType] = true
+			} else if policyType == vt.AllVerifySuccess {
+				// if policy is 'all'
+				if !castedReport.IsSuccess {
```

Comment:
Wodnering if this statement will ever be hit? AS this should be caught in the ContinueVerifyOnFailure right?

Reply:
yeah... it won't be hit with the current executor set up. Should I remove it? I wasn't sure if the implementation of this method should be independent of its usage in the executor.